### PR TITLE
add first tests

### PR DESCRIPTION
**Pull request by stanleyycheung**
restructure file directory to convention

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jmanchuck) [inline .vscode/settings.json]: can we keep this one out of the repo it feels more like a local setting thing - we might have our own custom settings

File: .github/workflows/github-actions-demo.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: GitHub Actions Demo
```
Comment (jmanchuck) [inline .github/workflows/github-actions-demo.yml]: for next PR

File: .vscode/settings.json (hunk 1 above)
Comment (stanleyycheung) [reply .vscode/settings.json]: yes - should be gitignore but somehow left out

File: .github/workflows/github-actions-demo.yml (hunk 2 above)
Comment (stanleyycheung) [reply .github/workflows/github-actions-demo.yml]: i'll make this working with pytest then merge

File: tests/test_limiter.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            return request
+
+        for i in range(100):
+            mock_function(i)
+
+        start = 0
+        for i in range(1, len(called_times)):
+            if called_times[i] - called_times[start] >= datetime.timedelta(0, 1):
+                assert i - start - 1 <= 20
+                start = i
```
Comment (jmanchuck) [inline tests/test_limiter.py]: Feel like this one might miss some cases - say we have [t0... t19] >= 1, but 17 of those requests [t3... t19] happen in the last 0.1 seconds. We jump start index to 19, then e.g. [t19... t38] >= 1 but 10 of those are in the first 0.3 seconds of this window - we end up with [t3... t29] in 0.5 seconds. Very edge case though.

tldr; I think sliding window of 1 second width might be more accurate

File: Pipfile.lock [hunk 4]
```diff
@@ -0,0 +1,427 @@
+{
```
Comment (jmanchuck) [inline Pipfile.lock:1]: Putting comment cuz idk what this is. Looks like autogenerated file do we need this one for dependencies?

**Review by jmanchuck — COMMENTED** [review]
Lockfile and small comment on rate limiter test

File: Pipfile.lock (hunk 4 above)
Comment (stanleyycheung) [reply Pipfile.lock:1]: yeah its auto generated by `pipenv` so just don't touch it

File: tests/test_limiter.py (hunk 3 above)
Comment (stanleyycheung) [reply tests/test_limiter.py]: agreed. sliding window is a bit fiddly though

**Review by jmanchuck — APPROVED** [review]
